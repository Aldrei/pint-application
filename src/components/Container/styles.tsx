import styled from 'styled-components';

export const ContainerStyle = styled.div`
  width: 100%;
  padding: 20px;
  border: 1px solid red;
`;
