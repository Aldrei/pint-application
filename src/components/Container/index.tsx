import React from 'react';

import { ContainerStyle } from './styles';

export interface Props {
  children?: React.ReactNode
}

const Container = (props: Props): React.ReactElement => {
  return (
    <ContainerStyle className='containerComp'>{props.children}</ContainerStyle>
  );
};

export default Container;