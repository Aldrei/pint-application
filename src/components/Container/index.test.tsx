import { render } from '@testing-library/react';
import React from 'react';

import Container from './index';

describe('Container component', () => {
  it('Should render correctly', () => {
    const nodeEl = render(<Container><h1 className="child-test">Test</h1></Container>)

    const nodeElChild = nodeEl.baseElement.getElementsByClassName('child-test');
    expect(nodeElChild.length).toBe(1);
    
    expect(nodeEl.baseElement).toMatchSnapshot();
  })
})