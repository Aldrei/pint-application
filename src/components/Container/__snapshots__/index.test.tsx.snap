// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container component Should render correctly 1`] = `
<body>
  <div>
    <div
      class="containerComp MuiBox-root css-1cy5oc"
    >
      <h1
        class="child-test"
      >
        Test
      </h1>
    </div>
  </div>
</body>
`;
